tst/com/amazon/corretto/crypto/provider/test/FipsStatusTest.java

```diff
@@ -0,0 +1,41 @@
+package com.amazon.corretto.crypto.provider.test;
+
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+import com.amazon.corretto.crypto.provider.FipsStatusException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+import javax.crypto.KeyGenerator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(TestResultLogger.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock(value = TestUtil.RESOURCE_GLOBAL, mode = ResourceAccessMode.READ)
+public class FipsStatusTest {
+
+    private final AmazonCorrettoCryptoProvider provider = AmazonCorrettoCryptoProvider.INSTANCE;
+
+    @Test
+    public void givenAccpBuiltWithFips_whenAWS_LC_fips_failure_callback_expectException() throws Exception {
+        if (provider.isFips()) {
+            assertTrue(provider.isFipsStatusOk());
+            assertNotNull(KeyGenerator.getInstance("AES", provider));
+            // call the failure callback
+            NativeTestHooks.callAwsLcFipsFailureCallback();
+            assertFalse(provider.isFipsStatusOk());
+            // we should not be able to get any service object
+            assertThrows(FipsStatusException.class, () -> KeyGenerator.getInstance("AES", provider));
+            // we need to flip the status back to OK so the rest of tests would work
+            NativeTestHooks.flipFipsStatus();
```

Comment:
perhaps add one more `assertNotNull(KeyGenerator.getInstance("AES", provider));` on L39 to ensure we can get instances now?

Reply:
Once the flag is false, it remains false. I added a new target in our CMake for this test so we can run it independent of the others and not to worry about flipping the flag back for the sake of the other tests.